machine/v1beta1/types_azureprovider.go

```diff
@@ -215,6 +240,56 @@ type OSDisk struct {
 	CachingType string `json:"cachingType,omitempty"`
 }
 
+// DataDisk specifies the parameters that are used to add one or more data disks to the machine.
+// A Data Disk is a managed disk that's attached to a virtual machine to store application data.
+// It differs from an OS Disk as it doesn't come with a pre-installed OS, and it cannot contain the boot volume.
+type DataDisk struct {
```

Comment:
I think we will want to add some notes to this to say that you need to add ignition configuration to a customer user data secret to perform the disk partition/format/mount operations, so it's clear users need to do that per disk. Can also add a note that the LUN will be needed to coordinate between the disk and its mount point on the host

Reply:
Yes, good point. I was precisely doing that. Here it is: https://github.com/openshift/api/pull/1119/commits/b6963011c5947ae2bab1c22c6589efde95cc8c99